components/script/dom/globalscope.rs

```diff
@@ -3626,11 +3626,7 @@ impl GlobalScope {
                 // Step 3.1: If target is not null, and global is a Window,
                 // and target’s shadow-including root is not global’s associated Document, set target to null.
                 if let Some(window) = self.downcast::<Window>() {
-                    if !window
-                        .Document()
-                        .upcast::<Node>()
-                        .is_shadow_including_inclusive_ancestor_of(event_target.upcast())
-                    {
+                    if event_target.upcast::<Node>().owner_document() != window.Document() {
```

Comment:
Might be nice to put a comment here explaining why the owner document is always the target's shadowing-including root.